mleap-core/src/main/scala/ml/combust/mleap/core/types/Casting.scala

```diff
@@ -184,13 +184,41 @@ object Casting {
                   val cc = c.asInstanceOf[(Any) => ByteString]
                   (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(cc).toArray)
               }
+            }
+          }
+        }.orElse {
```

Comment:
When the base cast returns `None`, we just want to reshape the list into a tensor without changing the types of each element